src/NetEscapades.EnumGenerators/EnumGenerator.cs

```diff
@@ -40,9 +40,21 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             .SelectMany(static (m, _) => m!.Value)
             .WithTrackingName(TrackingNames.InitialExternalExtraction);
 
+        // Check for enums in generic types and generate diagnostics
+        IncrementalValuesProvider<Diagnostic> diagnostics = context.SyntaxProvider
+            .ForAttributeWithMetadataName(Attributes.EnumExtensionsAttribute,
+                predicate: static (node, _) => node is EnumDeclarationSyntax,
+                transform: GetDiagnosticForGenericTypeEnum)
+            .Where(static diagnostic => diagnostic is not null)
+            .Select(static (diagnostic, _) => diagnostic!)
+            .WithTrackingName(TrackingNames.GenericTypeDiagnostics);
```

Comment:
We should not be doing this inside a source generator - it should be written completely separately as a standard analyzer